### PR TITLE
Fix misuse of overly verbose log levels and unify use of log and tracing

### PR DESCRIPTION
**Pull request by ten3roberts**
This uses the appropriate log levels for our logging. 

The recommendation is:
`info` is surfaced to, and used as a means to communicate with the users, `debug` is intended for lower priority and implementation oriented messages, and trace is for very verbose and detailed information, most commonly includes debugged structs, or very implementation specific information that is almost entirely useful for internal devs and never users.

Additionally, this means the prefixed log filtering that were a common source of surprising behavior has been almost entirely removed. The only remaining targets are external crates that don't use logging appropriately.

The split usage of log, and its superset replacement *tracing* has been fixed, and tracing is used throughout. Tracing has lot of more useful features, and is especially good at capturing spatially temporal information through the use of spans. 

Tracing also has support for further detailed and structured logging, async, and much more, and is what is the new standard in the wider rust ecosystem.

Fixes: #584 
Fixes: #459

### CODE REVIEW

**Comment by ten3roberts** [comment]
# Old

image

**Note**: this uses our meticulously crafted specifc filtering rules that makes *enabling* additional logs very inconvenient as you can't differentiate between logs

# New

image

File: app/src/client/wasm.rs [hunk 1]
```diff
@@ -15,21 +15,16 @@ pub fn initialize(
...
-                level,
-                "[{name}] {prefix}: {}",
-                message.strip_suffix('\n').unwrap_or(message)
-            );
+            match ty {
+                MessageType::Info => tracing::info!(%module_name, "{}", message),
+                MessageType::Warn => tracing::warn!(%module_name, "{}", message),
+                MessageType::Error => tracing::error!(%module_name, "{}", message),
+                MessageType::Stdout => tracing::info!(%module_name, "stdout {}", message),
+                MessageType::Stderr => tracing::info!(%module_name, "stderr: {}", message),
```
Comment (pierd) [inline app/src/client/wasm.rs:26]: Any reason why stderr has `:` and stdout doesn't?

**Review by pierd — APPROVED** [review]

File: app/src/client/wasm.rs (hunk 1 above)
Comment (pierd) [reply app/src/client/wasm.rs:26]: Also maybe let's use error level for stderr

**Comment by ten3roberts** [comment]
Update: 

A slightly more compact format

image

Additionally, there is a feature flag `tracing-tree` `cargo run --features tracing-tree -- run ...` that makes it *much* clearer which logs come from which scope and how they are grouped both spatially and temporally.

image

**Comment by ten3roberts** [comment]
I mean, just *look* at this 😁 

image

This is so useful for debugging contexts and knowing how logs related to each other

**Comment by ten3roberts** [comment]
This `<unknown time>` is fixed as well

image

**Pull request merged**